cognite/client/_api/data_sets.py

```diff
@@ -0,0 +1,247 @@
+        limit: int = None,
+    ) -> Generator[Union[DataSet, DataSetList], None, None]:
+        """Iterate over data sets
+
+        Fetches data sets as they are iterated over, so you keep a limited number of data sets in memory.
+
+        Args:
+            chunk_size (int, optional): Number of data sets to return in each chunk. Defaults to yielding one data set a time.
+            metadata (Dict[str, str]): Custom, application-specific metadata. String key -> String value.
+            created_time (Union[Dict[str, Any], TimestampRange]): Range between two timestamps.
+            last_updated_time (Union[Dict[str, Any], TimestampRange]): Range between two timestamps.
+            external_id_prefix (str): Filter by this (case-sensitive) prefix for the external ID.
+            write_protected (bool): No description.
+            limit (int, optional): Maximum number of data sets to return. Defaults to 25. Set to -1, float("inf") or None
+                to return all items.
+
+        Yields:
+            Union[DataSet, DataSetList]: yields DataSet one by one if chunk is not specified, else DataSetList objects.
+        """
+        filter = DataSetFilter(
+            metadata=metadata,
+            created_time=created_time,
+            last_updated_time=last_updated_time,
+            external_id_prefix=external_id_prefix,
+            write_protected=write_protected,
+        ).dump(camel_case=True)
+        return self._list_generator(method="POST", chunk_size=chunk_size, filter=filter, limit=limit)
+
+    def __iter__(self) -> Generator[DataSet, None, None]:
+        """Iterate over data sets
+
+        Fetches data sets as they are iterated over, so you keep a limited number of data sets in memory.
+
+        Yields:
+            Event: yields DataSet one by one.
+        """
+        return self.__call__()
+
+    def create(self, data_set: Union[DataSet, List[DataSet]]) -> Union[DataSet, DataSetList]:
+        """`Create one or more data sets. <https://docs.cognite.com/api/v1/#operation/createDataSets>`_
+
+        Args:
+            data_set: Union[DataSet, List[DataSet]]: Data set or list of data sets to create.
+
+        Returns:
+            Union[DataSet, DataSetList]: Created data set(s)
+
+        Examples:
+
+            Create new data sets::
+
+                >>> from cognite.client import CogniteClient
+                >>> from cognite.client.data_classes import DataSet
+                >>> c = CogniteClient()
+                >>> data_sets = [DataSet(name="1st level"), DataSet(name="2nd level")]
+                >>> res = c.data_sets.create(data_sets)
+        """
+        return self._create_multiple(items=data_set)
+
+    def retrieve(self, id: Optional[int] = None, external_id: Optional[str] = None) -> Optional[DataSet]:
+        """`Retrieve a single data set by id. <https://docs.cognite.com/api/v1/#operation/getDataSets>`_
+
+        Args:
+            id (int, optional): ID
```

Comment:
We should have a better description for ID and IDs.

Reply:
agree. but it's the same for other resource types, so I were to change it I would do it for all of them
we can do it on this pr as well ofc